### PR TITLE
Add customMK Genesis Macro Pad

### PR DESCRIPTION
**Pull request by infinityis**
Adding keyboard and keymap files for customMK Genesis, a 5x4 macro pad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/custommk/genesis/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			RGB_HUI, RGB_HUD, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_ORTHO(
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[3] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/via/keymap.c:42]:
```suggestion
	[3] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			KC_P0,   KC_SPC,  KC_PDOT, KC_PENT),
+
+	[1] = LAYOUT_ORTHO(
+			KC_TRNS, RGB_TOG, RGB_MOD, KC_TRNS, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			RGB_VAI, RGB_VAD, RGB_SPI, RGB_SPD, 
+			KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+			RGB_HUI, RGB_HUD, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/via/keymap.c:35]:
```suggestion
	[2] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_ORTHO(
+			MO(1),   KC_PSLS, KC_PAST, KC_PMNS, 
+			KC_P7,   KC_P8,   KC_P9,   KC_PPLS, 
+			KC_P4,   KC_P5,   KC_P6,   KC_PGUP, 
+			KC_P1,   KC_P2,   KC_P3,   KC_PGDN, 
+			KC_P0,   KC_SPC,  KC_PDOT, KC_PENT),
+
+	[1] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/via/keymap.c:28]:
```suggestion
	[1] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/via/keymap.c:21]:
```suggestion
	[0] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/default/keymap.c:21]:
```suggestion
	[0] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+			MO(1),   KC_PSLS, KC_PAST, KC_PMNS, 
+			KC_P7,   KC_P8,   KC_P9,   KC_BSPC, 
+			KC_P4,   KC_P5,   KC_P6,   KC_PPLS, 
+			KC_P1,   KC_P2,   KC_P3,   KC_CAPS,
+			KC_P0,   KC_SPC,  KC_PDOT, KC_ENT),
+
+	[1] = LAYOUT_ORTHO(
```
Comment (zvecr) [inline keyboards/custommk/genesis/keymaps/default/keymap.c:28]:
```suggestion
	[1] = LAYOUT_ortho_5x4(
```

File: keyboards/custommk/genesis/info.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Genesis", 
+    "url": "https://www.customMK.com", 
+    "maintainer": "customMK", 
+    "width": 4, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/custommk/genesis/info.json:8]: This doesnt cover the 2 layouts that are defined for the keyboard

File: keyboards/custommk/genesis/genesis.h [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	K40, K41, K42, K43  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   KC_NO }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   KC_NO }, \
+	{ K40,   KC_NO, K42,   K43 }  \
+}
+
+#define LAYOUT_ORTHO( \
```
Comment (zvecr) [inline keyboards/custommk/genesis/genesis.h:35]:
```suggestion
#define LAYOUT_ortho_5x4( \
```

File: keyboards/custommk/genesis/genesis.c [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "genesis.h"
+
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (zvecr) [inline keyboards/custommk/genesis/genesis.c:19]:
```suggestion
__attribute__((weak)) void encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/custommk/genesis/config.h [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+#define ENCODERS_PAD_A { F6, D2 }
+#define ENCODERS_PAD_B { F7, D1 }
+#define ENCODER_RESOLUTION 4
+#define TAP_CODE_DELAY 10
+
+#define BOOTMAGIC_LITE_COLUMN 0
+#define BOOTMAGIC_LITE_ROW 0
+
```
Comment (zvecr) [inline keyboards/custommk/genesis/config.h:57]:
```suggestion

```

File: keyboards/custommk/genesis/genesis.h [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+	K20, K21, K22, K23, \
+	K30, K31, K32, K33, \
+	K40, K41, K42, K43  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   KC_NO }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   KC_NO }, \
+	{ K40,   KC_NO, K42,   K43 }  \
+}
```
Comment (zvecr) [inline keyboards/custommk/genesis/genesis.h:33]:
```suggestion
#define LAYOUT_numpad_5x4( \
	K00, K01, K02, K03, \
	K10, K11, K12,      \
	K20, K21, K22, K23, \
	K30, K31, K32,      \
	  K40,    K42, K43  \
) { \
	{ K00,   K01,   K02,   K03 }, \
	{ K10,   K11,   K12,   KC_NO }, \
	{ K20,   K21,   K22,   K23 }, \
	{ K30,   K31,   K32,   KC_NO }, \
	{ K40,   KC_NO, K42,   K43 }  \
}
```

File: keyboards/custommk/genesis/rules.mk [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+ENCODER_ENABLE = yes
+RGBLIGHT_ENABLE = yes
```
Comment (zvecr) [inline keyboards/custommk/genesis/rules.mk:20]:
```suggestion

# Bootloader selection
BOOTLOADER = qmk-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by infinityis** [comment]
Thank you! All suggested changes have been incorporated.

File: keyboards/custommk/genesis/config.h [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+#pragma once
+
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xF35B
+#define PRODUCT_ID      0xFAB0
+#define MANUFACTURER    customMK
+#define DEVICE_VER      0x0001
```
Comment (fauxpark) [inline keyboards/custommk/genesis/config.h:26]:
```suggestion
#define DEVICE_VER      0x0001
#define MANUFACTURER    customMK
```

File: keyboards/custommk/genesis/readme.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Supported: Genesis Macro Pad
+* Hardware Availability: [customMK](https://shop.custommk.com/products/genesis_macropad)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make custommk/genesis:default
+
+Flashing example for this keyboard:
+
+    make custommk/genesis:default:dfu
```
Comment (fauxpark) [inline keyboards/custommk/genesis/readme.md:18]:
```suggestion
    make custommk/genesis:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by infinityis — COMMENTED** [review]
Recommended suggestions have been incorporated

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**